stacker_blueprints/eks.py

```diff
@@ -0,0 +1,427 @@
+    Role
+)
+
+from troposphere.policies import (
+    AutoScalingReplacingUpdate,
+    AutoScalingRollingUpdate,
+    UpdatePolicy
+)
+
+from troposphere import eks
+from stacker.blueprints.base import Blueprint
+
+
+class Cluster(Blueprint):
+    VARIABLES = {
+        "Name": {
+            "type": str,
+            "description": "The name of the cluster to create.",
+        },
+        "ExistingRoleArn": {
+            "type": str,
+            "description": "IAM Role ARN with EKS assume role policies. One "
+                           "will be created if it's not provided.",
+            "default": ""
+        },
+        "Version": {
+            "type": str,
+            "description": "Kubernetes version",
+            "default": "",
+        },
+        "SecurityGroupIds": {
+            "type": str,
+            "description": "A comma separated list of security group ids for "
+                           "controlling ENI access from EKS to workers",
+        },
+        "SubnetIds": {
+            "type": str,
+            "description": "A comma separated list of subnet ids where you "
+                           "will launch your worker nodes",
+        }
+    }
+
+    # This creates an IAM role which EKS requires, as described here:
+    # https://docs.aws.amazon.com/eks/latest/userguide/getting-started.html#eks-create-cluster
+    def create_iam_role(self):
+        eks_service_role_id = "EksServiceRole"
+        t = self.template
+        t.add_resource(
+            Role(
+                eks_service_role_id,
+                AssumeRolePolicyDocument=Policy(
+                    Statement=[
+                        Statement(
+                            Effect=Allow,
+                            Action=[AssumeRole],
+                            Principal=Principal("Service",
+                                                ["eks.amazonaws.com"])
+                        )
+                    ]
+                ),
+                Path="/",
+                ManagedPolicyArns=[
+                    "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy",
+                    "arn:aws:iam::aws:policy/AmazonEKSServicePolicy",
+                ]
+            )
+        )
+        return GetAtt(eks_service_role_id, "Arn")
+
+    def get_iam_role(self):
+        role_arn = self.get_variables()["ExistingRoleArn"]
+        if role_arn:
+            return role_arn
+        return self.create_iam_role()
+
+    def create_template(self):
+        t = self.template
+        role_arn = self.get_iam_role()
+        variables = self.get_variables()
+
+        args = {}
+        version = variables["Version"]
+        if version:
+            args["Version"] = version
+
+        # This is a fully qualified stacker name, prefixed with the namespace.
+        eks_name_tag = self.context.get_fqn(variables["Name"])
+
+        t.add_resource(
+            eks.Cluster(
+                "EksCluster",
+                Name=eks_name_tag,
```

Comment:
In general, it's usually a good idea to avoid `Name`s in Cloudformation, due to limitaitons with Name Types in Cloudformation (they can't be updated with "requires replacement" updates without changing the Name: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-name.html)

I'm not 100% certain this is a Name Type, but I'm pretty sure it is.

Reply:
It is a Name type. It's required for an EKS cluster, can't skip it.

Reply:
Are you sure?  It doesn't look like it's required (I don't think name types are ever required): https://github.com/cloudtools/troposphere/blob/master/troposphere/eks.py#L20